client/me/get-apps/mobile-download-card.jsx

```diff
@@ -28,16 +28,31 @@ import {
 } from 'state/account-recovery/settings/selectors';
 import getUserSettings from 'state/selectors/get-user-settings';
 import hasUserSettings from 'state/selectors/has-user-settings';
-import { sendSMS } from 'state/mobile-download-sms/actions';
+import { http } from 'state/data-layer/wpcom-http/actions';
 import phoneValidation from 'lib/phone-validation';
 import userAgent from 'lib/user-agent';
 import twoStepAuthorization from 'lib/two-step-authorization';
 
-import {
-	isAppSMSRequestSending,
-	didAppSMSRequestCompleteSuccessfully,
-	didAppSMSRequestCompleteWithError,
-} from 'state/selectors/get-apps-sms-request';
+function sendSMS( phone ) {
```

Comment:
I wonder if it makes more sense to move this to [state/data-getters/index.js](https://github.com/Automattic/wp-calypso/blob/259ccb40468b770928f92e9f7bba870dc899d417/client/state/data-getters/index.js). We already have a precedent for a `POST` request there - see `createAutoDraft()`.

Reply:
The purpose of the `requestHttpData` and the data-getters that use it is to cache the result: GET the data only once and return the cached data on repeated requests. That doesn't make much sense with `POST`.

`createAutoDraft` is a Gutenlypso utility that's no longer used and will be removed soon.

When requesting the SMS, we always want the request to go to the server. The extra `requestHttpData` layer is not needed.

Reply:
Makes sense to me 👍

Reply:
for what it's worth, `requestHttpData()` _was_ actually designed with `POST` requests in mind. the purpose isn't to cache but rather to simplify the interface for defining data that comes from HTTP requests. Even though we're trying to trigger some remote action here we're still making a request for that data.

in the case where we don't want to use any data from the return we would still need to store it in order to make that system work, but I don't think the consequences would be that bad.

the fact that this is a side-effect only option makes `http()` as written a good candidate, but for kicks and giggles here's the general way I might do it with `requestHttpData()`

```js
const sendSMS = phone => requestHttpData(
	'sms-code-has-been-sent',
	http( { method: 'POST', …, path: '/me/get-apps/send-download-sms', … } )
);

const Sender = connect(
	() => ( {
		smsStatus: ( {
			'success': 'sent',
			'failure': 'failed',
			'pending': 'sending',
			'uninitialized': 'not sent',
		} )[ getHttpData( 'sms-code-has-been-sent' ).status ]
	} ),
	{
		sendSMS,
		successNotice,
		errorNotice
	}
)( class extends Component {
	componentDidUpdate( { smsStatus: oldStatus } ) {
		const { smsStatus: newStatus } = this.props;

		if ( newStatus === 'sent' && oldStatus !== 'sent' ) {
			successNotice( … );
		} else if ( newStatus === 'failed' && oldStatus !== 'failed' ) {
			errorNotice( … );
		}
	}	
} );
```

so here we have maybe easier data fetching logic but we move the state transition logic into the UI component even though it's a kind of global side-effect. given that this is probably specific to this very component I don't see that as a terrible thing but I probably would tend to leave it in the more generic data layer as @jsnajdr did because that's specifically the area for side effects and the notice logic is coupled with the data where it belongs, not with the UI